compaction/compaction.cc

```diff
@@ -268,15 +268,23 @@ struct compaction_writer {
         : compaction_writer(nullptr, std::move(writer), std::move(sst)) {}
 };
 
+struct splitting_partition_tag { };
+using splitting_partition = bool_class<splitting_partition_tag>;
+
 class compacted_fragments_writer {
     compaction& _c;
     std::optional<compaction_writer> _compaction_writer = {};
-    using creator_func_t = std::function<compaction_writer(const dht::decorated_key&)>;
+    using creator_func_t = std::function<compaction_writer(const dht::decorated_key&, splitting_partition)>;
     using stop_func_t = std::function<void(compaction_writer*)>;
     creator_func_t _create_compaction_writer;
     stop_func_t _stop_compaction_writer;
     std::optional<utils::observer<>> _stop_request_observer;
     bool _unclosed_partition = false;
+    struct partition_state {
+        dht::decorated_key_opt dk;
+        ::tombstone tombstone;
```

Comment:
Good, but please add a comment why we need to duplicate the tombstone.

Reply:
Sure.